x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -2250,6 +2250,24 @@ public void testLookupJoinIndexMode() {
         assertThat(e.getMessage(), containsString("1:25: invalid [test] resolution in lookup mode to an index in [standard] mode"));
     }
 
+    public void testLookupJoinRemoteClusterUnsupported() {
+        assumeTrue("requires LOOKUP JOIN capability", EsqlCapabilities.Cap.JOIN_LOOKUP_V11.isEnabled());
+
+        var remoteIndexResolution = loadMapping("mapping-default.json", "remote:my-index");
+        var lookupResolution = AnalyzerTestUtils.defaultLookupResolution();
+        VerificationException e = expectThrows(
+            VerificationException.class,
+            () -> analyze(
+                "FROM remote:my-index | LOOKUP JOIN languages_lookup ON language_code",
```

Comment:
I'd add test cases with multiple lookups, and test cases where more stuff happens between the `FROM` and the `LOOKUP JOIN`.